src/sqlfluff/core/linter/linter.py

```diff
@@ -116,6 +116,11 @@ def _load_raw_file_and_config(
         # Return the raw file and config
         return raw_file, file_config, encoding
 
+    @staticmethod
+    def _normalise_newlines(string: str) -> str:
+        """Normalise newlines to unix-style line endings."""
+        return regex.sub(r"\n|\r\n|\r", "\n", string)
```

Comment:
Do we need this first one? Seems to be replacing `"\n"` with `"\n"`?

```suggestion
        return regex.sub(r"\r\n|\r", "\n", string)
```

Reply:
strictly no, but thought it was clearer to read as replacing all newlines with the unix newline (i.e. normalise), although in this case it's not like we'll ever change the newline replace style so happy either way.

I'll let you commit the suggestion and merge if you want